modules/ld/sparql-client/src/main/scala/ch/epfl/bluebrain/nexus/commons/sparql/client/SparqlClient.scala

```diff
@@ -118,21 +118,23 @@ class SparqlClient[F[_]](sparqlBase: Uri)(implicit
   /**
     * Checks if a index exists as a namespace
     *
-    * @param name the name of the index
-    * @return ''Left(())'' when the index does not exist, ''Right(())'' when it does exist
+    * @param index the name of the index
+    * @return ''false'' when the index does not exist, ''true'' when it does exist
     *         and it signals an error otherwise.
     */
-  def indexExists(name: String): F[Either[Unit, Unit]] = {
-    cl(Get(s"$sparqlBase/namespace/$name")).flatMap { resp =>
-      val discardedEntity = cl.discardBytes(resp.entity)
+  def exists(index: String): F[Boolean] = {
+    val req = Get(s"$sparqlBase/namespace/$index")
+    cl(req).flatMap { resp =>
       resp.status match {
         case StatusCodes.OK       =>
-          discardedEntity.map(_ => Right(()))
+          cl.discardBytes(resp.entity).map(_ => true)
```

Comment:
Since there's no need to wait for the discard bytes to complete, you can just call `cl.discardBytes` and `F.pure` independently.

Reply:
True. I just saw you approved and I didn't see the comment. Now it has been already merged. However I can change that in another commit